integration-cli/docker_cli_pull_test.go

```diff
@@ -266,12 +266,12 @@ func (s *DockerHubPullSuite) TestPullClientDisconnect(c *check.C) {
 func (s *DockerSuite) TestPullLinuxImageFailsOnWindows(c *check.C) {
 	testRequires(c, DaemonIsWindows, Network)
 	_, _, err := dockerCmdWithError("pull", "ubuntu")
-	assert.ErrorContains(c, err, "no matching manifest")
+	assert.ErrorContains(c, err, "no matching manifest for windows")
 }
 
 // Regression test for https://github.com/docker/docker/issues/28892
 func (s *DockerSuite) TestPullWindowsImageFailsOnLinux(c *check.C) {
 	testRequires(c, DaemonIsLinux, Network)
-	_, _, err := dockerCmdWithError("pull", "microsoft/nanoserver")
-	assert.ErrorContains(c, err, "cannot be used on this platform")
+	_, _, err := dockerCmdWithError("pull", "mcr.microsoft.com/windows/servercore:ltsc2019")
```

Comment:
When I try this with docker-ce 19.03.0-beta1 on a linux machine:
```
$ docker pull mcr.microsoft.com/windows/servercore:ltsc2019
ltsc2019: Pulling from windows/servercore
no matching manifest for linux/amd64 in the manifest list entries
```

However, when I try this on docker for mac with docker-ce 18.09.3 engine:
```
$ docker pull mcr.microsoft.com/windows/servercore:ltsc2019
ltsc2019: Pulling from windows/servercore
no matching manifest for unknown in the manifest list entries
```

Reply:
I think this PR is good to go if expecting the arch in the error message is the right thing to expect.

Reply:
Seeing the same on Docker for Mac, so looks like there might be an issue there (or perhaps something we fixed since?)

Reply:
PR https://github.com/moby/moby/pull/38574 that fixes that unknown message doesn't have a 18.09 tag, so it comes with 19.03

Reply:
Ah, right! I knew there was one; let me have a look at backporting it to 18.09

Reply:
Backport is up; https://github.com/docker/engine/pull/197